src/embassy/time_driver.rs

```diff
@@ -0,0 +1,394 @@
+impl ChannelsAccess for crate::pac::timer1::RegisterBlock {
+    type Counter = u32;
+
+    fn set_chn_cc_value(&self, channel: usize, value: Self::Counter) {
+        match channel {
+            0 => self.ch0cv().write(|w| w.ch0val().bits(value)),
+            1 => self.ch1cv().write(|w| w.ch1val().bits(value)),
+            2 => self.ch2cv().write(|w| w.ch2val().bits(value)),
+            3 => self.ch3cv().write(|w| w.ch3val().bits(value)),
+            _ => return,
+        }
+    }
+
+    fn set_chn_ie(&self, channel: usize, value: bool) {
+        match channel {
+            0 => self.dmainten().modify(|_, w| w.ch0ie().bit(value)),
+            1 => self.dmainten().modify(|_, w| w.ch1ie().bit(value)),
+            2 => self.dmainten().modify(|_, w| w.ch2ie().bit(value)),
+            3 => self.dmainten().modify(|_, w| w.ch3ie().bit(value)),
+            _ => return,
+        };
+    }
+}
+
+#[cfg(feature = "time-driver-tim14")]
+impl ChannelsAccess for crate::pac::timer14::RegisterBlock {
+    type Counter = u16;
+
+    fn set_chn_cc_value(&self, channel: usize, value: Self::Counter) {
+        match channel {
+            0 => self.ch0cv().write(|w| w.ch0val().bits(value)),
+            1 => self.ch1cv().write(|w| w.ch1val().bits(value)),
+            _ => return,
+        }
+    }
+
+    fn set_chn_ie(&self, channel: usize, value: bool) {
+        match channel {
+            0 => self.dmainten().modify(|_, w| w.ch0ie().bit(value)),
+            1 => self.dmainten().modify(|_, w| w.ch1ie().bit(value)),
+            _ => return,
+        };
+    }
+}
+
+// Clock timekeeping works with something we call "periods", which are time intervals
+// of 2^15 ticks. The Clock counter value is 16 bits, so one "overflow cycle" is 2 periods.
+// Timer1 has 32 bits counter, however for a sake of consistency, only 16 bits are used.
+//
+// A `period` count is maintained in parallel to the Timer hardware `counter`, like this:
+// - `period` and `counter` start at 0
+// - `period` is incremented on overflow (at counter value 0)
+// - `period` is incremented "midway" between overflows (at counter value 0x8000)
+//
+// Therefore, when `period` is even, counter is in 0..0x7FFF. When odd, counter is in 0x8000..0xFFFF
+// This allows for now() to return the correct value even if it races an overflow.
+//
+// To get `now()`, `period` is read first, then `counter` is read. If the counter value matches
+// the expected range for the `period` parity, we're done. If it doesn't, this means that
+// a new period start has raced us between reading `period` and `counter`, so we assume the `counter` value
+// corresponds to the next period.
+//
+// `period` is a 32bit unsigned integer, so It overflows on 2^32 * 2^15 / 32768 seconds of uptime, which is 136 years.
+fn calc_now(period: u32, counter: u32) -> u64 {
+    ((period as u64) << 15) + ((counter ^ ((period & 1) << 15)) as u64)
+}
+
+const ALARM_STATE_NEW: AlarmState = AlarmState::new();
```

Comment:
Why this constant, rather than just calling `AlarmState::new()` where you need it? Having both seems like unneccesary duplication.